### PR TITLE
Run the command using users default shell

### PR DESCRIPTION
**Pull request by Itai-Nelken**
I tested this, only problem I encountered is a segfault if the SHELL environment variable doesn't exist, if it exists but is empty, the program takes care of it:
**when the shell is detected succesfully:**
```bash
itai@pinebookpro:~/DollarSkip$ ./temp echo 123
/bin/bash -c 'echo 123'
123
itai@pinebookpro:~/DollarSkip$
```
**else its run using the default for `system` (`sh`):**
```bash
itai@pinebookpro:~/DollarSkip$ ./temp echo 123
echo 123
123
itai@pinebookpro:~/DollarSkip$
```
for some reason the updated install script commit that you already merged is appearing here, I have no idea why.

### CODE REVIEW

**Comment by CleanMachine1** [comment]
I don't see the problem. Whats the issue?

**Comment by CleanMachine1** [comment]
Do all shells use -c for commands?

**Comment by Itai-Nelken** [comment]
> Do all shells use -c for commands?

I tested with `bash` and `sh`, and according to the `dash` man page it also uses `-c`. I'm not sure about `fish` and that is all the shells I know about.

**Comment by CleanMachine1** [comment]
Is there any gain using the users shell?

**Comment by CleanMachine1** [comment]
Ignore that.

**Comment by Itai-Nelken** [comment]
I don't understand, ignore what?

**Comment by Itai-Nelken** [comment]
all the shells I know of (`sh`, `bash`, `dash`, and `fish`) use `-c` for running a command.

**Comment by bl-ue** [comment]
`zsh` too 💡

**Comment by bl-ue** [comment]
You could also consider using `execve` to replace run the specified executable and replace the current process with that one.
This works (but you have to specify the absolute path to executables, and if an error occurs no shell message is printed, `execve` just returns `-1`):
```c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main(int argc, char **argv, char **envp)
{
    if (argc == 1)
        return 0;
    execve(argv[1], argv + 1, envp);
}
```

**Comment by bl-ue** [comment]
**tl;dr** don't use `execve`, use `system` and auto-detect the shell (however you can)

---

But I'm rethinking [that](https://github.com/CleanMachine1/DollarSkip/pull/13#issuecomment-855295622) — using a shell with `system` is actually better because it saves the hassle of passing the env like that, and it also uses a subshell.  The problem as @Itai-Nelken points out is that `system` doesn't necessarily use the user's current shell.

Take for example the following scenario.

A user is running zsh, and they're looking at some zsh repo, like https://github.com/ohmyzsh/ohmyzsh, and on that repo they're told to run some special zsh-only commands to effect this is or that:

>Do this or that:
>```
>$ zle foo bar
>$ zle bar foo
>```

I'll run it now, once without and once with the dollar sign, in zsh: 

```console
❯ zle foo bar
zle: widgets can only be called when ZLE is active

❯ $ zle foo bar
sh: zle: command not found
```
You can see the difference.  That's because `sh`, which is what `system()`'s executing, doesn't have the zsh-only-builtin `zle`.

**Comment by CleanMachine1** [comment]
Ok.

I didn't want to publish code which isn't prefect.

So I compiled yours but instead added stdio.h and instead of executing the code i printed it with `printf("%s", command);`



```bash

./a.out this is a test

-c '/bin/bash -c 'this is a test'


``` 

This is alarming

**Comment by CleanMachine1** [comment]
Things start to get broken since the `shell` variable is set to 12 and is empty.

I starts to break it. I recommend we set it to 128 in size and put `= {0}` at the end of it.

**Comment by CleanMachine1** [comment]
All seems to fix its self once I add change some aspects of the variable

`char shell[128] = {0}`

**Comment by Itai-Nelken** [comment]
I used this code:
```bash
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
int main(int argc, char **argv)
{
	char command[8096] = {0};
	if(argc>1) {
		char shell[10], *env=getenv("SHELL"); //get the nev var 'SHELL'
		if(strcmp(env, "")) { //if env != ""
			strcpy(shell, env); //copy env to shell
			strcat(shell, " -c \'"); //add " -c '"
			strcat(command, shell); //add it to command
			for (int i=1; i<argc; i++){
				strcat(strcat(command,argv[i])," ");
			} //add all the command (cmd args) to command
			command[strlen(command) - 1] = '\0';
			strcat(command, "\'");// add "'"
		} else {
			for (int i=1; i<argc; i++){
				strcat(strcat(command,argv[i])," ");
			}
			command[strlen(command) - 1] = '\0';
		}
		//system(command);// run the command
		printf("%s\n", command);
	}
}
```
and I ran: 
```
itai@pinebookpro:~/DollarSkip$ ./temp this is a test
/bin/bash -c 'this is a test'
itai@pinebookpro:~/DollarSkip$ 
```
so the output you got is weird @CleanMachine1.

**Comment by CleanMachine1** [comment]
Let me have another look and me do some debugging to see where it goes wrong.
printf to the rescue :)

**Comment by CleanMachine1** [comment]
Would you like me to move this PR to a branch here?

**Comment by CleanMachine1** [comment]
Actually I am going to do it. I need to do some testing easier. I may change it back when I am done or just open a PR to merge it to main at a later date.

**Comment by CleanMachine1** [comment]
Merging.

**Comment by Itai-Nelken** [comment]
@bl-ue If every command that was run would go through the DollarSkip, I would care more about shell specific commands, but for now, I think its not something that matters much and is more of a nice extra feature that can be added. maybe we can modify the `system` function to use the detected default shell? I don't believe that would be to hard.

**Comment by CleanMachine1** [comment]
@Itai-Nelken Could you please make a gitter account, maybe. I would like talk a little bit more privately. Feel free to decline.

Sign up is really easy since its all linked with github.

**Comment by Itai-Nelken** [comment]
@CleanMachine1 I created a account, how do I get you?

**Comment by CleanMachine1** [comment]
i didn't see bl-ues comments, let me have another good read.

**Comment by bl-ue** [comment]
@Itai-Nelken try visiting https://gitter.im/CleanMachine1 ;)

**Comment by Itai-Nelken** [comment]
@bl-ue thanks!
